pyclibrary/c_parser.py

```diff
@@ -60,193 +67,212 @@ def win_defs(version='1500'):
     version : unicode
         Version of the MSVC to consider when parsing.
 
+    force_update : bool, optional
+        If True, the cached header files are reparsed, even if they are
+        up-to-date
+
+    sdk_dir : str
+        Has to refer to the visual studio header files directory, where
+        all windows header files are located in.
+
     Returns
     -------
-    parser : CParser
+    parser : MSVCParser
         CParser containing all the infos from te windows headers.
 
     """
-    header_files = ['WinNt.h', 'WinDef.h', 'WinBase.h', 'BaseTsd.h',
-                    'WTypes.h', 'WinUser.h']
-    if not CParser._init:
-        logger.warning('Automatic initialisation : OS is assumed to be win32')
-        from .init import auto_init
-        auto_init()
-    d = os.path.dirname(__file__)
-    p = CParser(
-        header_files,
-        macros={'_WIN32': '', '_MSC_VER': version, 'CONST': 'const',
-                'NO_STRICT': None, 'MS_WIN32': ''},
-        process_all=False
-        )
+    if sdk_dir is None:
+        sdk_dir = []
+    dir = os.path.dirname(__file__)
+    cache_file_name = os.path.join(dir, 'headers', 'WinDefs.cache')
+
+    parser = MSVCParser(header_dirs=[sdk_dir],
+                        predef_macros={'_WIN32': '', 'NO_STRICT': ''},
+                        msc_ver=version, arch=32, )
```

Comment:
Unfortunately: Yes!

There are some ```#ifdefs``` in the windows headers, that check for one of the architecture defines (```_M_IX86```, ```_M_AMD64```, ...). Honsestly said, I do not know if the difference is relevant for us, but I could imagine, that there are corner cases, where it **is** relevant if you specified a platform define or not.

In the next commit the architecture will even be configurable (32 or 64 bit), although in most cases this should make no difference...